### PR TITLE
Editor: Added video transcoding buttons to design menu

### LINKED ISSUE #10634: UI Simplification: Floating menu: Video options
## Feature Description

Make the video options elements in the floating design menu functional.

Design: [Figma](https://www.figma.com/file/dc9xrVlRhQq4KTE9WQzU1R/_Editor---Sprint-Feature-Level-Designs_?node-id=11394%3A457235)
Menus: Video
Optional rendering: Trim only renders for local videos, Mute only renders for local, non-muted videos.
Notes: This includes two elements: Trim, and Mute. Note that all the functionality already exists in the current video options panel and should be refactored. The mute button graphics should be double-checked with UX before implementation (we have a mute icon we could use).

EDIT: The Loop checkbox has been moved into a separate issue and description updated accordingly: https://github.com/GoogleForCreators/web-stories-wp/issues/10693

## Additional information

Screen Shot 2022-02-18 at 19 03 42

### PR DESCRIPTION
**Pull request by barklund**
## Context

This makes the buttons for mute and trim video functional.

Note that this also re-frames the mute icon inside its SVG file to allow it to be used for an icon button similarly to other icons. The result in the library looks the same as before.

## To-do

- [x] New mute icon ~?~
- [x] Tests

## Non-user-facing changes

| Video type | Screenshot |
|-|-|
| Local video with audio | Screen Shot 2022-03-03 at 12 38 47 |
| Local video without audio | Screen Shot 2022-03-03 at 12 32 46 |
| Non-local video | Screen Shot 2022-03-03 at 12 32 50 |

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10634

### CODE REVIEW

File: packages/story-editor/src/components/floatingMenu/elements/mute.js
Comment (miina) [inline packages/story-editor/src/components/floatingMenu/elements/mute.js]:
```suggestion
      disabled={isDisabled || isMuting}
```

File: packages/story-editor/src/components/floatingMenu/elements/trim.js
Comment (miina) [inline packages/story-editor/src/components/floatingMenu/elements/trim.js]:
```suggestion
      disabled={isDisabled || isTrimming}
```

File: packages/story-editor/src/components/floatingMenu/elements/mute.js
Comment (miina) [reply packages/story-editor/src/components/floatingMenu/elements/mute.js]: Hm, this is not related to this PR but in case of some videos, I get this console error:
Screenshot 2022-03-03 at 11 29 22

Unfortunately I'm not able to repro it consistently. @spacedmonkey is this error something you've seen recently?

**Review by miina — APPROVED** [review]
LGTM (pending the better icon / tests) but would be great if @spacedmonkey could also review.

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (spacedmonkey) [inline packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: @timarney Will this line need to be changed?

File: packages/story-editor/src/components/floatingMenu/elements/mute.js
Comment (spacedmonkey) [reply packages/story-editor/src/components/floatingMenu/elements/mute.js]: I was not able to replicate that. Can provide steps to replicate or a video or something.

Comment (swissspidy) [reply packages/story-editor/src/components/floatingMenu/elements/mute.js]: @miina Perhaps the original unmuted video got deleted in the meantime?

**Comment by spacedmonkey** [comment]
I have marked this ticket as blocked by https://github.com/googleforcreators/web-stories-wp/issues/10641. As changed there effect this PR. I am also looping in @timarney, who worked on this ticket.

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (swissspidy) [reply packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: Yes, the function is being renamed in that other PR

File: packages/story-editor/src/components/floatingMenu/elements/mute.js
Comment (miina) [reply packages/story-editor/src/components/floatingMenu/elements/mute.js]: @swissspidy Hm, I thought that issue got fixed somewhere 🤔 Can't find it currently though for muting, saw just this for trimming: https://github.com/GoogleForCreators/web-stories-wp/issues/10245

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (spacedmonkey) [reply packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: And use the element id.

**Comment by swissspidy** [comment]
Just because we're renaming a function doesn't mean it's _blocked_ by that PR..? 🤔

**Comment by spacedmonkey** [comment]
> Just because we're renaming a function doesn't mean it's blocked by that PR..? 

To be clear, we are removing the trimming function and replacing it with another one. This new function requires element id to be passed to. I am just ensuring that element id is available in the in the new hook and there isn't going to be an issue here.

**Comment by merapi** [comment]
Not this PR fault, but I can't trim my local video, anybody has a clue? @spacedmonkey @swissspidy 

image
image

**Comment by spacedmonkey** [comment]
@merapi It seems like cors headers are not setup correctly. I see the url of your site is stories-bravo.local. What are you using for local development? I am using not the docker setup. Have you tried looking at the site using https?

File: packages/design-system/src/icons/muted.svg
Comment (barklund) [inline packages/design-system/src/icons/muted.svg:1]: This new icon is pretty big, bitwise. I did optimize it as much as possible using SVGO, but it's still annoyingly large. Could be handcrafted to be much smaller, but I'm not going to bother with it.

File: packages/story-editor/src/components/floatingMenu/elements/mute.js
Comment (barklund) [reply packages/story-editor/src/components/floatingMenu/elements/mute.js]: Hmm, this seems unrelated to this PR. I'm not seeing any issues. I fixed the original comment here in 0c0d1a483a and will mark as resolved for now.

File: packages/story-editor/src/components/floatingMenu/elements/trim.js
Comment (barklund) [reply packages/story-editor/src/components/floatingMenu/elements/trim.js]: Fixed in 0c0d1a483a.

**Comment by barklund** [comment]
> Not this PR fault, but I can't trim my local video, anybody has a clue? @spacedmonkey @swissspidy

You have to be on HTTPS for transcoding to work - even locally.

**Comment by spacedmonkey** [comment]
> You have to be on HTTPS for transcoding to work - even locally.

Yes, it has always been like that. Either https or localhost as the domain. Works fine with docker env and [local](https://localwp.com/). How have you tested trimming the past?

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (timarney) [reply packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: yes
https://github.com/GoogleForCreators/web-stories-wp/pull/10712/files#diff-d581131a68dd353c626d1714ea3b4d65115979020ed22b61c7e5d4a0a8fed0ddR96

**Comment by barklund** [comment]
> > You have to be on HTTPS for transcoding to work - even locally.
> 
> Yes, it has always been like that. Either https or localhost as the domain. Works fine with docker env and [local](https://localwp.com/). How have you tested trimming the past?

Sorry for the confusion @spacedmonkey, I should have explicitly addressed @merapi in my comment.

My comment (_"You have to be on HTTPS for transcoding to work - even locally"_) was a reply to@merapi about why he was not able to use transcoding on his setup, not a question to anyone.

**Comment by merapi** [comment]
Thanks for the clues, indeed it was about `https`, most of the time I test on `http://localhost` so it works there even without a secure connection.

**Review by merapi — APPROVED** [review]
👌🏻

**Review by spacedmonkey — CHANGES_REQUESTED** [review]
This PR needs to be tweaked now that https://github.com/GoogleForCreators/web-stories-wp/pull/10712 is merged.

**Comment by barklund** [comment]
@timarney, @spacedmonkey, I've merged in from main and tried to generalize the functionality in the video manipulation panel into a hook, that could be reused. Please test this out and check if video transcoding (muting and trimming) works correctly when invoked both from Design Panel and the new Design Menu (to see the latter, enable feature flag for Floating menu).

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (barklund) [inline packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: @timarney, @spacedmonkey: I'm getting [an error for this one in karma](https://github.com/GoogleForCreators/web-stories-wp/runs/5421645159?check_suite_focus=true#step:7:837)? Can this not be run in karma or? Can the function at least exist even if not doing the right thing?

**Comment by timarney** [comment]
> Please test this out and check if video transcoding (muting and trimming) works correctly when invoked both from Design Panel and the new Design Menu 

I'll take a look on Mon.

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (swissspidy) [reply packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: @barklund The function is now called `isElementTriming` and accepts an element ID.

Comment (spacedmonkey) [reply packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: See https://github.com/GoogleForCreators/web-stories-wp/pull/10784/files#r818585757 and https://github.com/GoogleForCreators/web-stories-wp/pull/10712 for context.

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (spacedmonkey) [inline packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: This hook could be very useful for [VideoOptionsPanel](https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/story-editor/src/components/panels/design/videoOptions/videoOptions.js). In fact is mostly a repeat of that code. Why not make this code more reusable and use it in both places? It will make future changes and refactors much easier.

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (spacedmonkey) [inline packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: I think this maybe should be broken up into actions and state. It will make clearer to know what is state and actions.

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (barklund) [reply packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: That's exactly what I had done and wanted you to review. And forgot to push 🤦 .

So, here it is (for real this time). Sorry for the inconvenience!

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (barklund) [reply packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: I have now done that in 77bff9e with tests fixed in 62c72f9.

File: packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js
Comment (swissspidy) [reply packages/story-editor/src/components/floatingMenu/elements/shared/useVideoTranscoding.js]: Any thoughts on using a less generic name? At first glance `useVideoTranscoding` could easily be mistaken for `useFFmpeg` (in fact, that is its former name).

Maybe even `useVideoElementTranscoding` is an improvement, but I'm sure there are better names.

File: packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js
Comment (spacedmonkey) [inline packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: Feels like this hook should exist [here](https://github.com/GoogleForCreators/web-stories-wp/tree/main/packages/story-editor/src/app/media/utils). It is used for multiple commpents and should not live with any component.

File: packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js
Comment (spacedmonkey) [inline packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: Name of this hook feels a little generic here. Can we do better? @swissspidy any thoughts?

**Comment by spacedmonkey** [comment]
Other than some naming things and location of files, this looks good. I will approve after these changes are made.

File: packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js
Comment (barklund) [reply packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: The hook uses functionality from the story-editor package, so I didn't see an easy way to move to another package. We could extract that functionality, but that would make the hook less reusable (as the removed logic would have to be duplicated).

File: packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js
Comment (barklund) [reply packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: I like @swissspidy's suggestion of `useVideoElementTranscoding`. I was struggling a bit with naming here, have no better suggestions.

`useTrimOrMute`? 🤷

File: packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js
Comment (swissspidy) [reply packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: Jonny pointed to `packages/story-editor/src/app/media/utils`, not another package. But not sure it's a great fit.

File: packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js
Comment (swissspidy) [reply packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: Let's use `useVideoElementTranscoding` for now then, even though it's a bit long

Comment (barklund) [reply packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: Done in b1f31cba21.

File: packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js
Comment (barklund) [reply packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: Ah yes, I see now. Yeah, that could work. I have no strong opinions either way. It's only ever (according to current plans at least) going to be used in the two locations where it is now, so anything that works for those are ok with me.

Comment (barklund) [reply packages/story-editor/src/components/panels/design/videoOptions/useVideoTranscoding.js]: Moved in fc958e66c4.

**Review by spacedmonkey — APPROVED** [review]

**Pull request merged**
